doc/api/events.md

```diff
@@ -935,13 +935,351 @@ if the `EventEmitter` emits `'error'`. It removes all listeners when
 exiting the loop. The `value` returned by each iteration is an array
 composed of the emitted event arguments.
 
+## `EventTarget` and `Event` API
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+The global `EventTarget` and `Event` objects are a Node.js-specific adaptation
+of the [`EventTarget` Web API][] that are exposed by some Node.js core APIs.
+Neither the `EventTarget` nor `Event` classes are currently available for end
+user code to create.
+
+```js
+const target = getEventTargetSomehow();
+
+target.dispatchEvent({ type: 'foo' });
+
+target.addEventListener('foo', (event) => {
+  console.log('foo event happened!');
+});
+```
+
+The `EventTarget` class is similar in function to `EventEmitter` but evolved
+separately as part of browser DOM APIs. The Node.js implementation of
+`EventTarget` is an adaptation of the DOM API with a number of Node.js-specific
+extensions and behaviors intended to allow it to work more naturally within
+the Node.js environment.
+
+### Node.js `EventTarget` vs. DOM `EventTarget`
+
+There are four key differences between the Node.js `EventTarget` and the
+[`EventTarget` Web API][]:
+
+1. There is no concept of a hierarchical DOM and event propagation in Node.js.
+   That is, an event dispatched to an `EventTarget` does not propagate through
+   a hierarchy of nested target objects that may each have their own set of
+   handlers for the event.
+2. In the Node.js `EventTarget`, any object with a `type` property whose value
+   is a string may be dispatched as an `event`.
```

Comment:
That's a frustrating divergence that would lead to non-interoperable code.

Reply:
Understood. This was included specifically because userland implementations support it. Whether we end up supporting it here or not is still to be determined.

Reply:
@domenic how would you suggest node deals with shims and libraries?

Would a custom `[Symbol.event]` or something else be preferable as a protocol? 

Ideally I'd want something that both doesn't create a divergence and does allow third-party libraries to create `Event`s (like `dispatchEvent`ing an `Event` created from JSDom).

Reply:
The ability to support third-party polyfill type events is exactly the motivation here, fwiw. Especially since we're not exporting the `Event` constructor right away.

Reply:
I'd suggest exposing the `Event` constructor, so that libraries can write code that works in both Node and the DOM, instead of more Node-specific code. After all, if the goal is Node-specific code, `EventEmitter` is right there.

Reply:
Noted. We may get there eventually but for now the experimental implementation won't be exposed to user code so we have some wiggle room here.

Reply:
It'll be exposed via `AbortSignal.__proto__` right?

Reply:
If the goal is to be able to dispatch an `Event` from e.g. JSDOM, I don't see how that could work with requiring a specific constructor, since JSDOM would hardly use the global `Event` from Node (for isolation, and because it cannot implement all the DOM stuff JSDOM definitely needs). Exposing the constructor alone would not solve the interop problem

Reply:
That doesn't make sense as a goal, since JSDOM implements its own EventTarget and would purposefully not want to interoperate with Node's (as you said, for isolation).

Reply:
@jasnell I think I see Domenic's point here regarding branding, iiuc the API explicitly requires subclasses of Event to prevent this sort of interop. Supporting this would violate API guarantees.

Is it horrible? No, but it would make writing universal code harder and not easier probably.

I see value in adding interop when we have a compelling use case. If people feel strongly about not supporting it unless you feel strongly we should for the very least let's not document it in the meantime.

I also think it's fine to only support it in NodeEventEmitter with the extensions when the user does emit and not dispatchEvent for example.